errors/main.go

```diff
@@ -16,10 +16,30 @@ type TMError interface {
 	ABCILog() string
 }
 
+// New returns a new internal error instance. This function is defined so that
+// weave/errors package provides the same interface as standard library error
+// package.
+//
+// Current notation allows of additional code argument for compatibility with
+// an older errors package API. Once all code is migrated, it can be reduced to
+// a single case when only a description is provided.
+func New(description string, code ...uint32) error {
```

Comment:
q: Why do we need to be compatible with stdlib errors?
q:  Do we need this function at all? IMHO we should declare errors via `NewRootError` for our error codes or can just use `InternalErr.New("")` for legacy ones without a proper code defined

Reply:
agreed, we don’t need to be compatible to stdlib

Reply:
This change was made to makes the transition easier. If we won't use this function after the transition (`errors.InternalErr.New("...")` can be used instead), then `errors.New` can be removed.

Right now we sometimes use stdlib error, but because of the name conflict, we import it with an alias, for example as `stderrors`. I think it is nicer to just provide the same interface as the stdlib `errors` package does. 
I can see this being a redundant code, so I don't insist on having it. And if we use it, I don't mind having it either :man_shrugging: 

Should I keep it or remove already?

Reply:
Please remove